crates/wasmtime/src/runtime/func.rs

```diff
@@ -2256,14 +2269,6 @@ impl HostFunc {
     }
 }
 
-impl Drop for HostFunc {
-    fn drop(&mut self) {
-        unsafe {
-            self.engine.signatures().unregister(self.sig_index());
-        }
-    }
-}
```

Comment:
To confirm, `RegisteredType` is replacing this right? (IIRC we thought this didn't happen but looks like it does, but you wanted to refactor to make it more clear?)

Reply:
Correct and correct.

I think the clean up logic is a lot more clear now!